### ISSUE TITLE
Add retention for model snapshots

### ISSUE THREAD

**Issue by sean-zheng-amazon** [opendistro-for-elasticsearch/anomaly-detection#303]
**Is your feature request related to a problem? Please describe.**
Currently we don't have any retention set up for snapshots. This could cause disk space shortage especially for high cardinality case, where thousands of snapshots could be created in hours.

**Describe the solution you'd like**
Add retention mechanism to HC detectors, clean up the index for anything older than X days

**Describe alternatives you've considered**
Not an alternative, but something we'll do in parallel, is to reduce the snapshot size by sharing the common parts between models. We'll track this in another issue.

**Comment by kaituo** [comment]
We have checkpoint retention.  This PR https://github.com/opendistro-for-elasticsearch/anomaly-detection/commit/f0105ccfe6526bf045edb7f0b5ead2c13c166f4a starts deleting if checkpoint shard is larger than 50 GB regardless of retention period.
